app/controllers/practices_controller.rb

```diff
@@ -73,7 +73,16 @@ def remove_providers
   def destroy
     @practice = Practice.find(params[:id])
     Record.where(practice_id: @practice.id).delete
-    Provider.where(parent_id: @practice.provider.id).delete if @practice.provider
+    if @practice.provider
```

Comment:
Can some of this be replaced with dependent behavior (http://docs.mongodb.org/ecosystem/tutorial/ruby-mongoid-tutorial/#dependent-behaviour)? Maybe using nullify?